File: julia/Ccxt/test/base/test.urlencodeBase64.jl
using Test
using Ccxt
function testUrlencodeBase64()

    exchange = get(ccxt, Symbol("Exchange"), nothing)(Dict{Symbol, Any}(
        Symbol("id") => "sampleexchange"
    ));
    @test urlencodeBase64(exchange, "hello") == "aGVsbG8";
    @test urlencodeBase64(exchange, "hello world") == "aGVsbG8gd29ybGQ";
    @test urlencodeBase64(exchange, "test") == "dGVzdA";
    @test urlencodeBase64(exchange, "") == "";
    @test urlencodeBase64(exchange, "a") == "YQ";
    @test urlencodeBase64(exchange, "ab") == "YWI";
    @test urlencodeBase64(exchange, "abc") == "YWJj";
    @test urlencodeBase64(exchange, "abcd") == "YWJjZA";
    @test urlencodeBase64(exchange, "{\"user\":\"test\"}") == "eyJ1c2VyIjoidGVzdCJ9";
    @test urlencodeBase64(exchange, "subjects?_d") == "c3ViamVjdHM_X2Q";
    @test urlencodeBase64(exchange, "The quick brown fox") == "VGhlIHF1aWNrIGJyb3duIGZveA";
    @test urlencodeBase64(exchange, "123456789") == "MTIzNDU2Nzg5";
    binaryData = base16ToBinary(exchange, "191919191919");
    @test urlencodeBase64(exchange, binaryData) == "GRkZGRkZ";
end

File: ts/src/test/base/test.urlencodeBase64.ts


import assert from 'assert';
import ccxt from '../../../ccxt.js';

function testUrlencodeBase64 () {

    const exchange = new ccxt.Exchange ({
        'id': 'sampleexchange',
    });

    const inputs = [ 'hello', 'hello world', 'test', '', 'a', 'ab', 'abc', 'abcd', '{"user":"test"}', 'subjects?_d', 'The quick brown fox', '123456789' ];
    const expecteds = [ 'aGVsbG8', 'aGVsbG8gd29ybGQ', 'dGVzdA', '', 'YQ', 'YWI', 'YWJj', 'YWJjZA', 'eyJ1c2VyIjoidGVzdCJ9', 'c3ViamVjdHM_X2Q', 'VGhlIHF1aWNrIGJyb3duIGZveA', 'MTIzNDU2Nzg5' ];
    for (let i = 0; i < inputs.length; i++) {
        const value = inputs[i];
        const expected = expecteds[i];
        const result = exchange.urlencodeBase64 (value);
        assert (result === expected, 'urlencodeBase64 (' + value + ') != ' + expected);
    }
    const binaryData = exchange.base16ToBinary ('191919191919');
    assert (exchange.urlencodeBase64 (binaryData) === 'GRkZGRkZ');
}

export default testUrlencodeBase64;
